sdk/metric/internal/aggregate/sum.go

```diff
@@ -5,73 +5,82 @@ package aggregate // import "go.opentelemetry.io/otel/sdk/metric/internal/aggreg
-	values map[attribute.Distinct]sumValue[N]
 }
 
-func newValueMap[N int64 | float64](limit int, r func(attribute.Set) FilteredExemplarReservoir[N]) *valueMap[N] {
-	return &valueMap[N]{
-		newRes: r,
-		limit:  newLimiter[sumValue[N]](limit),
-		values: make(map[attribute.Distinct]sumValue[N]),
-	}
-}
-
-func (s *valueMap[N]) measure(ctx context.Context, value N, fltrAttr attribute.Set, droppedAttr []attribute.KeyValue) {
-	s.Lock()
-	defer s.Unlock()
-
-	v, ok := s.values[fltrAttr.Equivalent()]
-	if !ok {
-		fltrAttr = s.limit.Attributes(fltrAttr, s.values)
-		v = s.values[fltrAttr.Equivalent()]
-		v.res = s.newRes(fltrAttr)
-		v.attrs = fltrAttr
-	}
-
-	v.n += value
-	v.res.Offer(ctx, value, droppedAttr)
-
-	s.values[fltrAttr.Equivalent()] = v
+func (s *valueMap[N]) measure(
+	ctx context.Context,
+	value N,
+	fltrAttr attribute.Set,
+	droppedAttr []attribute.KeyValue,
+) {
+	sv := s.values.LoadOrStoreAttr(fltrAttr, func(attr attribute.Set) any {
+		return &sumValue[N]{
+			res:   s.newRes(attr),
+			attrs: attr,
+		}
+	}).(*sumValue[N])
+	sv.n.add(value)
+	// It is possible for collection to race with measurement and observe the
+	// exemplar in the batch of metrics after the add() for cumulative sums.
+	// This is an accepted tradeoff to avoid locking during measurement.
+	sv.res.Offer(ctx, value, droppedAttr)
 }
 
-// newSum returns an aggregator that summarizes a set of measurements as their
-// arithmetic sum. Each sum is scoped by attributes and the aggregation cycle
-// the measurements were made in.
-func newSum[N int64 | float64](monotonic bool, limit int, r func(attribute.Set) FilteredExemplarReservoir[N]) *sum[N] {
-	return &sum[N]{
-		valueMap:  newValueMap[N](limit, r),
+// newDeltaSum returns an aggregator that summarizes a set of measurements as
+// their arithmetic sum. Each sum is scoped by attributes and the aggregation
+// cycle the measurements were made in.
+func newDeltaSum[N int64 | float64](
+	monotonic bool,
+	limit int,
+	r func(attribute.Set) FilteredExemplarReservoir[N],
+) *deltaSum[N] {
+	return &deltaSum[N]{
 		monotonic: monotonic,
 		start:     now(),
+		hotColdValMap: [2]valueMap[N]{
+			{
+				values: limitedSyncMap{aggLimit: limit},
+				newRes: r,
+			},
+			{
+				values: limitedSyncMap{aggLimit: limit},
+				newRes: r,
+			},
+		},
 	}
 }
 
-// sum summarizes a set of measurements made as their arithmetic sum.
-type sum[N int64 | float64] struct {
-	*valueMap[N]
-
+// deltaSum is the storage for sums which resets every collection interval.
+type deltaSum[N int64 | float64] struct {
 	monotonic bool
 	start     time.Time
+
+	hcwg          hotColdWaitGroup
+	hotColdValMap [2]valueMap[N]
+}
+
+func (s *deltaSum[N]) measure(ctx context.Context, value N, fltrAttr attribute.Set, droppedAttr []attribute.KeyValue) {
+	hotIdx := s.hcwg.start()
+	defer s.hcwg.done(hotIdx)
+	s.hotColdValMap[hotIdx].measure(ctx, value, fltrAttr, droppedAttr)
 }
 
-func (s *sum[N]) delta(
+func (s *deltaSum[N]) delta(
```

Comment:
This can be renamed to `collect` now that the types are split.

```suggestion
func (s *deltaSum[N]) collect(
```